recipes/libfabric/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class LibfabricConan(ConanFile):
+    name = "libfabric"
+    description = "Open Fabric Interfaces"
+    topics = ("conan", "fabric")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://libfabric.org"
+    license = "BSD"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ['gni', 'psm', 'psm2', 'psm3', 'rxm', 'sockets', 'tcp', 'udp', 'usnic', 'verbs', 'bgq']
+    options = {
+        **{ p: "ANY" for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": "ANY",
+            "with_bgq_progress": [None, "auto", "manual"],
+            "with_bgq_mr": [None, "basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: "auto" for p in _providers },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "with_libnl": None,
+            "with_bgq_progress": None,
+            "with_bgq_mr": None
+        }
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def configure(self):
+        if self.settings.os == 'Windows' or self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The libfabric package cannot be built on Visual Studio.")
+        for p in self._providers:
+            if self.options.get_safe(p) not in ["auto", "yes", "no", "dl"] and not os.path.isdir(str(self.options.get_safe(p))):
```

Comment:
```suggestion
            if self.options.get_safe(p) not in ["auto", "yes", "no", "dl"] and not os.path.isdir(str(self.options.get_safe(p))):
```

I would prefer listing only what's allowed instead of ANY, it seems clear what is valid here, for instance `["auto", True, False, "dl"]`